src/os/macos.zig

```diff
@@ -25,43 +25,23 @@ pub fn appSupportDir(
     alloc: Allocator,
     sub_path: []const u8,
 ) AppSupportDirError![]u8 {
-    comptime assert(builtin.target.isDarwin());
-
-    const NSFileManager = objc.getClass("NSFileManager").?;
-    const manager = NSFileManager.msgSend(
-        objc.Object,
-        objc.sel("defaultManager"),
-        .{},
-    );
-
-    const url = manager.msgSend(
-        objc.Object,
-        objc.sel("URLForDirectory:inDomain:appropriateForURL:create:error:"),
-        .{
-            NSSearchPathDirectory.NSApplicationSupportDirectory,
-            NSSearchPathDomainMask.NSUserDomainMask,
-            @as(?*anyopaque, null),
-            true,
-            @as(?*anyopaque, null),
-        },
-    );
-
-    // I don't think this is possible but just in case.
-    if (url.value == null) return error.AppleAPIFailed;
-
-    // Get the UTF-8 string from the URL.
-    const path = url.getProperty(objc.Object, "path");
-    const c_str = path.getProperty(?[*:0]const u8, "UTF8String") orelse
-        return error.AppleAPIFailed;
-    const app_support_dir = std.mem.sliceTo(c_str, 0);
-
-    return try std.fs.path.join(alloc, &.{
-        app_support_dir,
+    return try makeCommonPath(alloc, .NSApplicationSupportDirectory, &.{
         build_config.bundle_id,
         sub_path,
     });
 }
 
+pub const CacheDirError = Allocator.Error || error{AppleAPIFailed};
+
+/// Return the path to the system cache directory with the given sub path joined.
+/// This allocates the result using the given allocator.
+pub fn cacheDir(
+    alloc: Allocator,
+    sub_path: []const u8,
+) CacheDirError![]u8 {
+    return try makeCommonPath(alloc, .NSCachesDirectory, &.{sub_path});
```

Comment:
There's a little bit of an inconsistency here in that `cacheDir` doesn't include the bundle ID (like `appSupportDir` does). That also means that we'll append just `ghostty` and not the bundle ID in the XDG path code. I think that's fine, and it's what some other cross-platform apps appear to do based on a quick look inside of my `~/Library/Caches` directory.

Reply:
Thanks for the thorough review! 

Regarding the bundle ID inconsistency: Looking at `~/Library/Caches`, I see both patterns in use - many apps use bundle IDs (like `com.microsoft.VSCode`, `com.googlecode.iterm2`) while some development tools use simpler names (like `deno`, `zls`). 

Since this PR focuses on fixing the immediate blocking issue with cache directories, I think we can keep the current behavior and revisit the bundle ID usage in a follow-up if needed.